examples/matching_engine/src/contract.rs

```diff
@@ -0,0 +1,608 @@
+                state_order.count.try_sub(cancel_count).unwrap()
+            }
+        };
+        let corr_cancel_count = state_order.count.try_sub(new_count).unwrap();
+        state_order.count = new_count;
+        Self::remove_zero_orders_from_level(view).await?;
+        Ok((corr_cancel_count, new_count == Amount::zero()))
+    }
+
+    async fn modify_order(
+        &mut self,
+        order_id: OrderId,
+        cancel_count: ModifyCount,
+    ) -> Result<Transfer, MatchingEngineError> {
+        let key_book = self.orders.get(&order_id).await?;
+        if let Some(key_book) = key_book {
+            match key_book.nature {
+                OrderNature::Bid => {
+                    let view = self.bids.load_entry_mut(&key_book.price.revert()).await?;
+                    let (cancel_count, remove_order_id) =
+                        Self::modify_order_level(view, order_id, cancel_count).await?;
+                    if remove_order_id {
+                        self.remove_order_id((key_book.owner, order_id)).await?;
+                    }
+                    let cancel_count0 = product_price_amount(key_book.price, cancel_count);
+                    let transfer = Transfer {
+                        owner: key_book.owner,
+                        amount: cancel_count0,
+                        token_idx: 0,
+                    };
+                    Ok(transfer)
+                }
+                OrderNature::Ask => {
+                    let view = self.asks.load_entry_mut(&key_book.price).await?;
+                    let (cancel_count, remove_order_id) =
+                        Self::modify_order_level(view, order_id, cancel_count).await?;
+                    if remove_order_id {
+                        self.remove_order_id((key_book.owner, order_id)).await?;
+                    }
+                    let transfer = Transfer {
+                        owner: key_book.owner,
+                        amount: cancel_count,
+                        token_idx: 1,
+                    };
+                    Ok(transfer)
+                }
+            }
+        } else {
+            Err(MatchingEngineError::OrderNotPresent)
+        }
+    }
+
+    async fn get_new_order_id(&mut self) -> Result<OrderId, MatchingEngineError> {
+        let value = self.next_order_number.get_mut();
+        let value_ret = *value;
+        *value += 1;
+        Ok(value_ret)
+    }
+
+    fn get_transfers(
+        nature: &OrderNature,
+        fill: Amount,
+        owner: &AccountOwner,
+        order: &OrderEntry,
+        price_level: Price,  // the price that was present in the level
+        price_insert: Price, // the price of the inserted order
+    ) -> Vec<Transfer> {
+        let mut transfers = Vec::new();
+        match nature {
+            OrderNature::Bid => {
+                // The order offers to buy token1 at price price_insert
+                // ---When the old order was created fill of token1 were commited
```

Comment:
We use markdown for comments: list items should use `* ` or `- ` for the "bullet".

Reply:
Ok.